### PR TITLE
Plane: fixed plane landing gear to obey LGR_OPTIONS bits

### PR DESCRIPTION
**Pull request by tridge**
we forced landing gear retract/deploy on takeoff and landing when we
should be following the options bits
thanks to Arace for pointing this out

### CODE REVIEW

File: ArduPlane/servos.cpp [hunk 1]
```diff
@@ -556,47 +556,24 @@ void Plane::set_servos_flaps(void)
...
+    if (control_mode == &mode_auto && hal.util->get_soft_armed() && is_flying() && gear.last_flight_stage != flight_stage) {
         switch (flight_stage) {
         case AP_Vehicle::FixedWing::FLIGHT_LAND:
-            cmd = AP_LandingGear::LandingGear_Deploy;
+            g2.landing_gear.deploy_for_landing();
             break;
         case AP_Vehicle::FixedWing::FLIGHT_NORMAL:
-            cmd = AP_LandingGear::LandingGear_Retract;
-            break;
-        case AP_Vehicle::FixedWing::FLIGHT_VTOL:
```
Comment (magicrub) [inline ArduPlane/servos.cpp:587]: what happens in VTOL?

**Comment by rmackay9** [comment]
I haven't checked the code but assuming it works, issue:https://github.com/ArduPilot/ardupilot/issues/13665 is resolved by this PR

File: ArduPlane/servos.cpp (hunk 1 above)
Comment (tridge) [reply ArduPlane/servos.cpp:587]: see the change in the quadplane code

**Comment by pompecukor** [comment]
@tridge @rmackay9 
This is still not working as it should.
LRG retracts on transition (Quad-plane) and ignores the altitude set in the parameters.

Very easy to reproduce in SITL of mission planner.

**Comment by pompecukor** [comment]
@tridge @rmackay9 

It is still not working, testing with options set to 0 or to only land. And it would still retract on takeoff.

In the below picture retract is set to 65m, Transition is set to 30m. Option is set to 0.
As you can see it still retracts.

image

**Comment by tridge** [comment]
>  As you can see it still retracts.

can you provide a SITL log? I just tested this in SITL and it worked as expected

**Comment by pompecukor** [comment]
ok, it was my fault. Was testing it with master, when it was not actually merged yet.

**Pull request merged**